Dockerfile

```diff
@@ -1,5 +1,5 @@
 # stage 1 Generate celestia-appd Binary
-FROM docker.io/golang:1.21.0-alpine3.17 as builder
+FROM docker.io/golang:1.20.0-alpine3.17 as builder
```

Comment:
[question] why does downgrading from Go 1.21 to Go 1.20 fix https://github.com/celestiaorg/celestia-app/issues/2249 ?

Reply:
I don't suspect it to be related to go itself but the docker image build. But, since I couldn't reproduce this locally, it only happens in CI, I suspect it has to do with the environment/architecture and the new docker image build

Reply:
Sorry I don't have enough information to approve. It looks like this Dockerfile upgraded from Go 1.20.7 to Go 1.21 in https://github.com/celestiaorg/celestia-app/pull/2232 which has successful CI checks for Docker...

<img width="519" alt="Screenshot 2023-08-11 at 12 16 47 PM" src="https://github.com/celestiaorg/celestia-app/assets/3699047/11f278b5-6cf0-4a5d-bf68-f63ec84f89bb">

Reply:
Sure. Anyway, after that, we added pyroscope dependency, which apparenlty needs some specific environment to build

Reply:
It was a good idea to investigate :pray: 

Here is the issue: https://github.com/grafana/pyroscope-golang/issues/38

I added a replace directive with the latest version of `godeltaprof` until they make a release that depends on it.

However, I am not sure whether this replace should be in core or in downstream libraries... What do you think? @rootulp @evan-forbes